src/util/inkscape.js

```diff
@@ -0,0 +1,221 @@
+const series = require('run-series')
+const uuid = require('uuid/v4')
+const os = require('os')
+const PNG = require('pngjs').PNG
+const semver = require('semver')
+const jsdom = require('jsdom')
+const { JSDOM } = jsdom
+const tinycolor = require('tinycolor2')
+
+const PATH_TO_BUILD = path.join(os.tmpdir(), 'orca-build')
+try {
+  fs.mkdirSync(PATH_TO_BUILD)
+} catch (e) {}
+
+const fillUrl = /(^|; )fill: url\('#([^']*)'\);/
+const fillRgbaColor = /(^|; )fill: rgba\(([^)]*)\);/
+const fillOpacityZero = /(^|\s*)fill-opacity: 0;/
+const strokeOpacityZero = /(^|; )stroke-opacity: 0;/
+const opacityZero = /(^|; )opacity: 0;/
+
+/** Node wrapper for Inkscape
+ *
+ * $ apt-get install inkscape
+ * ... or on OS X:
+ * $ brew install inkscape
+ *
+ * See:
+ * - https://linux.die.net/man/1/inkscape
+ * - https://en.wikipedia.org/wiki/Inkscape
+ */
+class Inkscape {
+  constructor (pathToInkscape) {
+    this.cmdBase = pathToInkscape || 'inkscape'
+  }
+
+  /** Convert SVG to EMF
+   *
+   * @param {buffer} svg : svg data buffer
+   * @param {object} opts
+   *  - id {string}
+   * @param {function} cb
+   *  - err {null || error}
+   *  - result {buffer}
+   */
+  svg2emf (svg, opts, cb) {
+    const id = opts.id || uuid()
+    const inPath = path.join(PATH_TO_BUILD, id + '-svg')
+    const outPath = path.join(PATH_TO_BUILD, id + '-emf')
+    const cmd = `${this.cmdBase} --file ${inPath} --export-emf ${outPath}`
+
+    const destroyTmpFiles = (cb) => parallel([
+      (cb) => fs.unlink(inPath, cb),
+      (cb) => fs.unlink(outPath, cb)
+    ], cb)
+
+    // Get background color from figure's definition
+    var bgColor
+    if (opts.figure.layout.paper_bgcolor) {
+      var color = tinycolor(opts.figure.layout.paper_bgcolor)
+      color = color.toRgb()
+      bgColor = [color.r, color.g, color.b]
+    } else {
+      bgColor = [255, 255, 255]
+    }
+    svg = this.cleanSvg(svg, bgColor)
+
+    series([
+      (cb) => fs.writeFile(inPath, svg, 'utf-8', cb),
+      (cb) => childProcess.exec(cmd, cb),
+      (cb) => fs.readFile(outPath, cb)
+    ], (err, result) => {
+      destroyTmpFiles(() => {
+        cb(err, result[2])
+      })
+    })
+  }
+
+  cleanSvg (svg, bgColor) {
+    const fragment = JSDOM.fragment(svg)
+
+    // Remove path and rectangles that are compleletely transparent
+    fragment.querySelectorAll('rect, path').forEach(function (node) {
+      var style = node.getAttribute('style')
+      if (style && (
+        (style.match(fillOpacityZero) && style.match(strokeOpacityZero)) ||
+        style.match(opacityZero)
+      )) node.remove()
+    })
+
+    // Set fill color to background color if its fill-opacity is 0 but stroke-opacity isn't 0
+    fragment.querySelectorAll('rect').forEach(function (node) {
+      var style = node.getAttribute('style')
+      if (!style) return
+      var m = style.match(fillOpacityZero)
+
+      if (m) {
+        var rgbFill = m[1] + `fill: rgb(${bgColor[0]},${bgColor[1]},${bgColor[2]});`
```

Comment:
:cow2: Let's make this one single ES6 string: `${m[1]} fill: rgb ... ;`

Reply:
Fixed in e76c52b